### PR TITLE
travis run to test travis api testing

### PR DESCRIPTION
**Pull request by bradymiller**
base code is from this PR:
https://github.com/openemr/openemr/pull/3453

GOAL:
Get the api travis tests working (note it is currently working when test on easy dev docker, so probably just a minor issue to sort out)

### CODE REVIEW

**Comment by bradymiller** [comment]
1st run failing api testing :)
(will play around with this more this evening; suspicious that maybe api server is not on as previously thought)

File: API_README.md [hunk 1]
```diff
@@ -4,6 +4,36 @@
...
+
+At a high level, the request processing flow consists of the following steps:
+```bash
+JSON Request -> Controller Component -> Validation -> Service Component -> Database
+```
+
+The logical response flow begins with the database result:
+```bash
+Database Result -> Service Component -> Controller Component -> RequestControllerHelper -> JSON Response
+```
```
Comment (stu01509) [inline API_README.md:28]: Just a little opinion, Maybe change the `bash` to `sh` will be more consistent with the current document :)

Comment (dixonwhitmire) [reply API_README.md:28]: @stu01509 I've updated this in #3453. Thanks!

**Comment by dixonwhitmire** [comment]
@bradymiller my latest updates to #3453 updates the `setAuthToken` function in the test client to return the full response object. I'm using that to [validate](https://github.com/openemr/openemr/pull/3453/files#diff-951ec28de3556f8b0be70b5906519d48R111) the auth process. 

This may give you more insight into the issue. If you can create a new branch based off of my recent changes, that is. Basically if that response object is coming back null than the API may not be enabled. I'll switch over to take a look this afternoon, time permitting.

Thanks!

File: API_README.md (hunk 1 above)
Comment (stu01509) [reply API_README.md:28]: I Got it 😀

**Comment by bradymiller** [comment]
hi @dixonwhitmire ,
I rebased this PR to your most current code and finally sorted the api issue out (was a simple permission issue; it was breaking since need write permissions to be able to store key for encryption). The commit you can cherry-pick is here (note I also added the needed `build_test_controllers` calls in travis):
https://github.com/openemr/openemr/pull/3459/commits/1c5713385e11a4f0d7da6dd84de622f8a7b7a042

**Comment by bradymiller** [comment]
@dixonwhitmire ,
Dang, spoke too early. The apache tests are fixed, but still gotta sort out the nginx tests.

**Comment by bradymiller** [comment]
time to resort to the insane dev environment :)

**Pull request closed without merging**